presto-main/src/main/java/com/facebook/presto/sql/planner/DeterminismEvaluator.java

```diff
@@ -47,7 +47,8 @@ protected Void visitFunctionCall(FunctionCall node, AtomicBoolean deterministic)
             // TODO: total hack to figure out if a function is deterministic. martint should fix this when he refactors the planning code
             if (node.getName().equals(QualifiedName.of("rand")) ||
                     node.getName().equals(QualifiedName.of("random")) ||
-                    node.getName().equals(QualifiedName.of("shuffle"))) {
+                    node.getName().equals(QualifiedName.of("shuffle")) ||
+                    node.getName().equals(QualifiedName.of("grouping"))) {
```

Comment:
According to Raghav the definition of non deterministic includes functions that should not be pushed down https://github.com/prestodb/presto/pull/5465#r70119024. But I agree that in general we shouldn't be doing this hardcoding of function names.